### PR TITLE
Fix for RetryUntil in In-memory executor

### PR DESCRIPTION
**Pull request by ashprakasan**

### CODE REVIEW

File: zio-flow/shared/src/main/scala/zio/flow/ZFlowExecutor.scala [hunk 1]
```diff
@@ -90,7 +90,7 @@ object ZFlowExecutor {
         state    <- stateRef.get
         vRef     <- ZIO.fromOption(state.getVariable(variableName))
         _        <- vRef.set(value.asInstanceOf)
-        _        <- stateRef.modify(state => (state.retry.forkDaemon, state.copy(retry = ZIO.unit))).flatten
+        //_ <- stateRef.modify(state => (state.retry.forkDaemon, state.copy(retry = ZIO.unit))).flatten
```
Comment (jdegoes) [inline zio-flow/shared/src/main/scala/zio/flow/ZFlowExecutor.scala:93]: Dead code?

Comment (ashprakasan) [reply zio-flow/shared/src/main/scala/zio/flow/ZFlowExecutor.scala:93]: That's a miss. I will delete the commented line.

Comment (ashprakasan) [reply zio-flow/shared/src/main/scala/zio/flow/ZFlowExecutor.scala]: The functions `getVariable` and `setVariable` in executor were not being used (Dead code). Deleted both.

File: zio-flow/shared/src/main/scala/zio/flow/ZFlowExecutor.scala [hunk 2]
```diff
@@ -274,9 +283,9 @@ object ZFlowExecutor {
 
             _ <- compile[I, E, timeout.ValueA](innerPromise, ref, input, flow)
             //TODO check other operations ensure done/to (promise) is called with await.
-//            _      <- status
-//                        .fold(p.succeed(None))(_ => innerPromise.await.timeout(duration).run.flatMap(e => p.done(e.map(Some(_)))))
-//                        .forkDaemon
+            //            _      <- status
+            //                        .fold(p.succeed(None))(_ => innerPromise.await.timeout(duration).run.flatMap(e => p.done(e.map(Some(_)))))
+            //                        .forkDaemon
```
Comment (jdegoes) [inline zio-flow/shared/src/main/scala/zio/flow/ZFlowExecutor.scala:269]: More dead code

File: zio-flow/shared/src/main/scala/zio/flow/ZFlowExecutor.scala [hunk 3]
```diff
@@ -405,7 +420,7 @@ object ZFlowExecutor {
     final case class State(
       tstate: TState,
       variables: Map[String, Ref[_]],
-      retry: UIO[Any] = ZIO.unit
+      retry: List[FlowPromise[_, _]] = List.empty
```
Comment (jdegoes) [inline zio-flow/shared/src/main/scala/zio/flow/ZFlowExecutor.scala:404]: Is having multiple promises well-defined here?

File: zio-flow/shared/src/main/scala/zio/flow/ZFlowExecutor.scala [hunk 4]
```diff
@@ -375,14 +387,17 @@ object ZFlowExecutor {
...
+                    p.await.run
+                      .flatMap(e => e.fold(cause => promise.halt(cause) as CompileStatus.Done, loop(_)))
                   else
                     p.await.run
                       .flatMap(e => e.fold(cause => promise.halt(cause), loop))
                       .forkDaemon as CompileStatus.Suspended
               } yield status
             )
       }
+      //} yield result
```
Comment (jdegoes) [inline zio-flow/shared/src/main/scala/zio/flow/ZFlowExecutor.scala:380]: Dead code

File: zio-flow/shared/src/main/scala/zio/flow/ZFlowExecutor.scala [hunk 5]
```diff
@@ -130,6 +131,11 @@ object ZFlowExecutor {
             (a, value2) = tuple
             _          <- vRef.set(value2)
             _          <- ref.update(_.addReadVar(vRef))
+            state      <- ref.get
+            _          <-
+              ZIO
+                .foreachPar_(state.retry)(fp => compile(fp.promise.asInstanceOf[Promise[Any, Any]], ref, (), fp.flow))
```
Comment (jdegoes) [inline zio-flow/shared/src/main/scala/zio/flow/ZFlowExecutor.scala:118]: If there is more than one promise in `state.retry`, it looks like we execute the flow multiple times. I think this is not desired behavior?

**Comment by vigoo** [comment]
Closing this because the in-memory executor is not working anymore.

**Pull request closed without merging**